src/libertem/viz.py

```diff
@@ -0,0 +1,349 @@
+
+
+class ColormapPerception(colors.LinearSegmentedColormap, Colormap2D):
+    """A perceptual colormap based on face-based luminance matching.
+
+    Based on a publication by Kindlmann et. al.
+    http://www.cs.utah.edu/~gk/papers/vis02/FaceLumin.pdf
+    This colormap tries to achieve an isoluminant perception by using a list of colors acquired
+    through face recognition studies. It is a lot better than the HLS colormap, but still not
+    completely isoluminant (despite its name). Also it appears a bit dark.
+
+    """
+
+    _log = logging.getLogger(__name__ + '.ColormapPerception')
+
+    CDICT = {'red': [(0/6, 0.847, 0.847),
+                     (1/6, 0.527, 0.527),
+                     (2/6, 0.000, 0.000),
+                     (3/6, 0.000, 0.000),
+                     (4/6, 0.316, 0.316),
+                     (5/6, 0.718, 0.718),
+                     (6/6, 0.847, 0.847)],
+
+             'green': [(0/6, 0.057, 0.057),
+                       (1/6, 0.527, 0.527),
+                       (2/6, 0.592, 0.592),
+                       (3/6, 0.559, 0.559),
+                       (4/6, 0.316, 0.316),
+                       (5/6, 0.000, 0.000),
+                       (6/6, 0.057, 0.057)],
+
+             'blue': [(0/6, 0.057, 0.057),
+                      (1/6, 0.000, 0.000),
+                      (2/6, 0.000, 0.000),
+                      (3/6, 0.559, 0.559),
+                      (4/6, 0.991, 0.991),
+                      (5/6, 0.718, 0.718),
+                      (6/6, 0.057, 0.057)]}
+
+    def __init__(self):
+        self._log.debug('Calling __init__')
+        super().__init__('perception', self.CDICT, N=256)
+        self._log.debug('Created ' + str(self))
+
+
+class ColormapHLS(colors.ListedColormap, Colormap2D):
+    """Colormap subclass for encoding directions with colors.
+
+    This class is a subclass of the :class:`~matplotlib.pyplot.colors.ListedColormap`
+    class. The class follows the HSL ('hue', 'saturation', 'lightness') 'Double Hexcone' Model
+    with the saturation always set to 1 (moving on the surface of the color
+    cylinder) with a lightness of 0.5 (full color). The three prime colors (`rgb`) are spaced
+    equidistant with 120° space in between, according to a triadic arrangement.
+    Even though the lightness is constant in the plane, the luminance (which is a weighted sum
+    of the RGB components which encompasses human perception) is not, which can lead to
+    artifacts like reliefs. Converting the map to a grayscale show spokes at the secondary colors.
+    For more information see:
+    https://vis4.net/blog/posts/avoid-equidistant-hsv-colors/
+    http://www.workwithcolor.com/color-luminance-2233.htm
+    http://blog.asmartbear.com/color-wheels.html
+
+    """
+
+    _log = logging.getLogger(__name__ + '.ColormapHLS')
+
+    def __init__(self):
+        self._log.debug('Calling __init__')
+        h = np.linspace(0, 1, 256)
+        l = 0.5 * np.ones_like(h)
```

Comment:
flake8 dislikes variable names that are ambiguous (`l` vs `I` etc.), thus the flake8 failure here. I don't feel strongly about this either way, we can either keep `l` as name and disable this check for this line with a `noqa` comment (see [flake8 documentation](http://flake8.pycqa.org/en/latest/user/violations.html#in-line-ignoring-errors) for details), or choose a different name.

Maybe for consistency disabling works best here?

Reply:
I'll add the corresponding comment!